### PR TITLE
Update to DYAMOND2 compset

### PR DESCRIPTION
**Pull request by crterai**
The PR adds the DYAMOND2 compset and applies some changes related to compsets relevant for DYAMOND simulations (outlined below). It also adds changes in PR#492 (https://github.com/E3SM-Project/scream/pull/492) by @brhillman . 

The changes are:

* Add F2010-SCREAM-LR-DYAMOND2 and F2010-SCREAM-HR-DYAMOND2 compsets
* Add and specify ne256 atmosphere IC file for DYAMOND1 and DYAMOND2
* Set macmic values that change with model resolution for 2000_scream_hr compsets
* Specify and use ifs provided SST files for DYAMOND1 and DYAMOND2 simulations. (note: SST files typically span 40 days and not longer)
* Add DY1 land IC file for 2010 forcings
* Add temporary DY2 land IC file for 2010 forcings
* Ensure land for 2000 forcings are set to sim_year 2000 and 2010 forcings are set to sim_year 2010
* Specify default use_case for 2010 forcing SCREAM cases.

### CODE REVIEW

File: components/cam/bld/namelist_files/use_cases/2000_scream_hr.xml [hunk 1]
```diff
@@ -65,8 +65,12 @@
 <deep_scheme             >off</deep_scheme>
 
 <!-- Macrophysics/microphysics coupling -->
-<cld_macmic_num_steps    > 6</cld_macmic_num_steps>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/use_cases/2000_scream_hr.xml:68]: Removing this line likely leaves no default setting if someone were to add a new grid that isn't listed below. Is this intentional, or is this case handled by the code itself?

File: components/cam/bld/namelist_files/use_cases/2010_scream_hr_dyamond2.xml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+<prescribed_aero_file nlev="128">mam4_0.9x1.2_L128_2000clim_c191106.nc</prescribed_aero_file>
+<prescribed_aero_cycle_yr>01</prescribed_aero_cycle_yr>
+
+<!-- Do radiation every five minutes for ne1024 -->
+<iradsw hgrid="ne1024np4"> 4 </iradsw>
+<iradlw hgrid="ne1024np4"> 4 </iradlw>
+
+<!-- Settings related to the SCREAM NH dycore -->
+<theta_hydrostatic_mode>.false.</theta_hydrostatic_mode>
+<tstep_type>9</tstep_type>
```
Comment (brhillman) [inline components/cam/bld/namelist_files/use_cases/2010_scream_hr_dyamond2.xml:177]: @mt5555 do we need to specify tstep_type here, or does setting theta_hydrostatic_mode = .false. automatically change the time stepping scheme?

File: components/elm/cime_config/config_component.xml [hunk 3]
```diff
@@ -83,6 +83,12 @@
       <value compset="SSP585_CAM5%CMIP6_ELM*_BGC%B">2015-2100_SSP585_CMIP6bgc_transient</value> 
       <value compset="2010_CAM5%CMIP6-LR_ELM">2010_CMIP6LR_control</value>
       <value compset="2010_CAM5%CMIP6-HR_ELM">2010_CMIP6HR_control</value>
+      <value compset="2010_CAM5.*SCREAM-LR" >2010_CMIP6_control</value>
+      <value compset="2010_CAM5.*SCREAM-HR" >2010_CMIP6_control</value>
+      <value compset="2010_CAM5.*SCREAM-LR-DYAMOND2" >2010_dy2scream_control</value>
```
Comment (brhillman) [inline components/elm/cime_config/config_component.xml:88]: This looks like it's using a new elm use_case file, but I don't see such a use case file added in this PR. Is this a mistake, or maybe the use case files didn't get committed?

File: components/cam/bld/namelist_files/use_cases/2000_scream_hr.xml (hunk 1 above)
Comment (crterai) [reply components/cam/bld/namelist_files/use_cases/2000_scream_hr.xml:68]: Good point about leaving in a default value. I'll add it back in in the next commit.

File: components/elm/cime_config/config_component.xml (hunk 3 above)
Comment (crterai) [reply components/elm/cime_config/config_component.xml:88]: Thanks for catching this. It did get left off the commit.

File: components/cam/bld/namelist_files/use_cases/2010_scream_hr_dyamond2.xml (hunk 2 above)
Comment (crterai) [reply components/cam/bld/namelist_files/use_cases/2010_scream_hr_dyamond2.xml]: I checked this and tried removing it at first, but realized that it defaults to tstep_type = 5. So I've added it back in.

**Comment by crterai** [comment]
Okay, I've made a few fixes following @brhillman's comments. There's also a fix to running conventional F2010-SCREAM ne4 and ne30 tests. I think the issues have been all fixed though. @brhillman  - would you review the PR again?

**Review by brhillman — APPROVED** [review]

**Pull request merged**